test/Sentry.Testing/EnvironmentVariableGuard.cs

```diff
@@ -3,21 +3,35 @@ namespace Sentry.Testing;
 public static class EnvironmentVariableGuard
 {
     // To allow different xunit collections use of this
-    private static readonly object Lock = new();
+    private static readonly SemaphoreSlim Lock = new(1, 1);
 
     public static void WithVariable(string key, string value, Action action)
     {
-        lock (Lock)
+        Lock.Wait();
+        Environment.SetEnvironmentVariable(key, value, EnvironmentVariableTarget.Process);
+        try
         {
-            Environment.SetEnvironmentVariable(key, value, EnvironmentVariableTarget.Process);
-            try
-            {
-                action();
-            }
-            finally
-            {
-                Environment.SetEnvironmentVariable(key, null, EnvironmentVariableTarget.Process);
-            }
+            action();
+        }
+        finally
+        {
+            Environment.SetEnvironmentVariable(key, null, EnvironmentVariableTarget.Process);
```

Comment:
I honest think if we have flaky tests due to env var, we have to take to be able to lock this env var read/write. Global state like this is often going to hunt us back

Reply:
On this one, I just didn't like the blocking `GetAwaiter().GetResult()` in the tests in `IntegrationMockedBackroundWorker`.  So I added `WithVariableAsync`, which uses a semaphore for locking since you can't `lock` around awaits.  The existing non-async was updated just to use the same semaphore.

But yeah, there's probably a way to do this without setting environment variables globally.